pkg/pmem-csi-operator/main.go

```diff
@@ -114,6 +115,24 @@ func Main() int {
 		return 1
 	}
 
+	list := &api.DeploymentList{}
+	if err := mgr.GetClient().List(context.TODO(), list); err != nil {
+		pmemcommon.ExitError("failed to get deployment list: %v", err)
+		return 1
+	}
+
+	activeList := []string{}
+	for _, d := range list.Items {
+		if d.DeletionTimestamp == nil {
+			activeList = append(activeList, d.Name)
+		}
+	}
+
+	if len(activeList) != 0 {
+		klog.Infof("There are active PMEM-CSI deployments (%v), hence not deleting the CRD.", activeList)
+		return 0
+	}
```

Comment:
Even with this check, there's still a race condition:
- a CR gets created that the operator hasn't seen yet
- operator shuts down and deletes CRD
- CR gets deleted

It's theoretical and probably not relevant in practice, but still...

For example, suppose an admin has a script which updates the operator and at the same time creates a CR. Now the chance of losing this race becomes much higher.

I'm not sure whether it's better to move CRD creation into the operator deployment. That may make it more complicated to update it. Let's leave it as it is.

Reply:
Yes, there is a chance of race condition and hard to cover. In fact, I expect the API server should protect the CRs and CRD deletion should fail in case of CRs exists. We are just trying to cover that.

Reply:
> I expect the API server should protect the CRs and CRD deletion should fail in case of CRs exists

Didn't we see the exact opposite? If the CRD was deleted, the CRs got destroyed. That's the issue you are trying to fix here (https://github.com/intel/pmem-csi/issues/579).